File: src/display.rs
use iced::{executor, Command, Container, Element, Length, Application, Settings, Image};

/// Display a file to the user, this can either be an image file or a text file
pub fn display(path: String) {
    Display::run(Settings::with_flags(path));
}

struct Display {
    path: String,
}

impl Application for Display {
    type Executor = executor::Null;
    type Message = ();
    type Flags = String;

    fn new(_flags: Self::Flags) -> (Display, Command<Self::Message>) {
        (Display { path: _flags }, Command::none())
    }

    fn title(&self) -> String {
        String::from("Template Searcher")
    }

    fn update(&mut self, _message: ()) -> Command<Self::Message> {
        Command::none()
    }

    fn view(&mut self) -> Element<()> {
        let img = Image::new(&self.path)
            .width(Length::Fill)
            .height(Length::Fill);

        Container::new(img)
            .width(Length::Fill)
            .height(Length::Fill)
            .padding(20)
            .center_x()
            .center_y()
            .into()
    }
}

File: src/display.rb
require "ruby2d"
require "fastimage"

class ImageDisplayer
    def new()
        set title: "Template Searcher"
        set diagnostics: true
        self.img = nil
    end

    def showImage(path)
        image_size = FastImage.size(path)

        set width: image_size[0]
        set height: image_size[1]
        
        # Delete the previous image
        self.img.remove unless img.nil? || img.empty?
        
        self.img = Image.new(path, width: image_size[0], height: image_size[1])
        show
    end

    def close()
        close
    end
end